File: test/core/test_owner.f90
program test_owner

use, intrinsic :: iso_fortran_env, only : stderr=>error_unit
use filesystem

implicit none

character(:), allocatable :: exe, user, name, group

valgrind : block

exe = getarg(0)
print '(a)', "Executable: " // exe

user = get_username()
print '(a)', "User: " // user

name = get_owner_name(exe)
print '(a)', "Owner name: " // name
if(len_trim(name) == 0) error stop "error: No owner information"

if(.not. is_windows()) then
  group = get_owner_group(exe)
  print '(a)', "Owner group: " // group
  if(len_trim(group) == 0) error stop "error: No group information"
endif

if(user == name) stop "OK: User and owner match"

end block valgrind

write(stderr, '(a)') "User and owner didn't match"

if(getenv("CI") == "true") then
  write(stderr, '(a)') "mismatched username and owner can happen on CI systems"
end if

end program

File: test/core/test_owner.cpp
#include <iostream>
#include <cstdlib>

#include "ffilesystem.h"
#include "ffilesystem_test.h"


int main(
#if __has_cpp_attribute(maybe_unused)
[[maybe_unused]]
#endif
int argc, char *argv[]) {

  std::string_view exe = argv[0];
  std::cout << "Executable: " << exe << "\n";

  const std::string user = fs_get_username();
  std::cout << "User: " << user << "\n";;

  const std::string name = fs_get_owner_name(exe);
  std::cout << "Owner name: " << name << "\n";;
  if (name.empty())
    err("No owner information");

  const std::string group = fs_get_owner_group(exe);
  std::cout << "Owner group: " << group << std::endl;
  if (group.empty())
    err("No owner group information");

  if (user == name) {
    std::cout << "OK: User and owner match\n";
    return EXIT_SUCCESS;
  }

  std::cerr << "User and owner didn't match\n";

  if (fs_getenv("CI") == "true")
    std::cerr << "mismatched username and owner can happen on CI systems\n";

  ok_msg("get_owner C++");

  return EXIT_SUCCESS;
}
